src/backend/egl/display.rs

```diff
@@ -61,6 +63,85 @@ pub struct EGLDisplay {
     logger: slog::Logger,
 }
 
+fn select_platform_display<N: EGLNativeDisplay + 'static>(
+    native: &N,
+    log: &::slog::Logger,
+) -> Result<*const c_void, Error> {
+    let dp_extensions = unsafe {
+        let p = wrap_egl_call(|| ffi::egl::QueryString(ffi::egl::NO_DISPLAY, ffi::egl::EXTENSIONS as i32))
+            .map_err(Error::InitFailed)?; //TODO EGL_EXT_client_extensions not supported
+
+        // this possibility is available only with EGL 1.5 or EGL_EXT_platform_base, otherwise
+        // `eglQueryString` returns an error
+        if p.is_null() {
+            return Err(Error::EglExtensionNotSupported(&["EGL_EXT_platform_base"]));
+        } else {
+            let p = CStr::from_ptr(p);
+            let list = String::from_utf8(p.to_bytes().to_vec()).unwrap_or_else(|_| String::new());
+            list.split(' ').map(|e| e.to_string()).collect::<Vec<_>>()
+        }
+    };
+    debug!(log, "Supported EGL client extensions: {:?}", dp_extensions);
+
+    for platform in native.supported_platforms() {
+        debug!(log, "Trying EGL platform: {}", platform.platform_name);
+
+        let log = log.new(o!("platform" => format!("{:?}", platform)));
+
+        let missing_extensions = platform
+            .required_extensions
+            .iter()
+            .filter(|ext| !dp_extensions.iter().any(|x| x == *ext))
+            .collect::<Vec<_>>();
+
+        if !missing_extensions.is_empty() {
+            info!(
+                log,
+                "Skipping EGL platform because one or more required extensions are not supported. Missing extensions: {:?}", missing_extensions
+            );
+            continue;
+        }
+
+        let display = unsafe {
+            wrap_egl_call(|| {
+                ffi::egl::GetPlatformDisplayEXT(
+                    platform.platform,
+                    platform.native_display,
+                    platform.attrib_list.as_ptr(),
+                )
+            })
+            .map_err(Error::DisplayCreationError)
+        };
+
+        let display = match display {
+            Ok(display) => {
+                if display == ffi::egl::NO_DISPLAY {
+                    info!(log, "Skipping platform because the display is not supported");
+                    continue;
+                }
+
+                display
+            }
+            Err(err) => {
+                info!(
+                    log,
+                    "Skipping platform because of an display creation error: {:?}", err
+                );
+                continue;
+            }
+        };
+
+        info!(
+            log,
+            "Successfully selected EGL platform: {}", platform.platform_name
+        );
+        return Ok(display);
+    }
+
+    crit!(log, "Unable to find suitable EGL platform");
```

Comment:
Because the user might have multiple gpus or different backends running at the same time, I would not describe this error as *critical*. smithay might very well be functional, so I would suggest using `error!` here instead.

Reply:
Right, makes sense :)
Done!